src/Google.Bigquery.V2/BigqueryClient.Queries.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Copyright 2016 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Apis.Bigquery.v2.Data;
+using System;
+
+namespace Google.Bigquery.V2
+{
+    public partial class BigqueryClient
+    {
+        /// <summary>
+        /// Executes a query synchronously.
+        /// </summary>
+        /// <param name="sql">The query in BigQuery's SQL dialect. Must not be null.</param>
+        /// <returns>The result of the query.</returns>
+        public virtual BigqueryResult ExecuteQuery(string sql)
+        {
+            throw new NotImplementedException();
+        }
+
+        // TODO: Do we want to parameterise for tableId etc? Note that this is the destination...
+        // (It may become part of an options object...)
+
+        /// <summary>
+        /// Creates a query job, optionally writing the results into a separate persitent table.
+        /// </summary>
+        /// <param name="sql">The query in BigQuery's SQL dialect. Must not be null.</param>
+        /// <param name="destinationTable">A destination table to write the results into. This may be
+        /// null, in which case the results will be stored in a temporary table.
+        /// </param>
+        /// <returns>The query job created. Use <see cref="GetQueryResult(JobReference)"/> to retrieve
+        /// the results of the query.</returns>
+        public virtual Job CreateQueryJob(string sql, TableReference destinationTable = null)
```

Comment:
Is a null default for destinationTable appropriate here?
Might it be more suitable for the user to need to explicitly pass null, so it's clear the choice has been made to us a temporary table?